docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,216 @@
+# Semantic Conventions for ASP.NEt Core specific metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for ASP.NET Core metrics, not specific to HTTP
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Routing](#routing)
+  * [Metric: `aspnet.routing.successful_matches`](#metric-aspnetroutingsuccessful_matches)
+  * [Metric: `aspnet.routing.failed_matches`](#metric-aspnetroutingfailed_matches)
+- [Exception metrics](#exception-metrics)
+  * [Metric: `aspnet.diagnostics_handler.exceptions`](#metric-aspnetdiagnostics_handlerexceptions)
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.queued_requests.duration`](#metric-aspnetrate_limitingqueued_requestsduration)
+  * [Metric: `aspnet.rate_limiting.rejected_requests`](#metric-aspnetrate_limitingrejected_requests)
+
+<!-- tocstop -->
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.successful_matches`
+
+**Status**: [Experimental][DocumentStatus]
+
+**TODO: can we try one aspnet.routing.matches** ?
+
+This metric is required.
```

Comment:
Are these docs auto-generated or replicating patterns found elsewhere? The .NET team isn't trying to define a new standard that others would implement so I wouldn't say we are requiring/recommending/suggesting anything. We care about standards in the sense that:
- We'd like to follow standards
- We'd like to document what we are doing so consumers of our metric data to know what to expect

If this is just the pattern for how these docs are produced thats np, its easy to ignore.

Reply:
I can remove or reword it.  Only tables are auto-generated, the rest is markdown.

However, if someone wants to contribute custom instrumentation for .net 7, classic asp.net, owin, etc, it'd be nice for them to know what they have to implement.

Reply:
updated the PR and resolving this thread.